cdap-ui/app/cdap/components/Replicator/Detail/TableScatterPlotGraph/scatterPlot.ts

```diff
@@ -158,8 +170,35 @@ export function renderScatterPlot(id: string, data: ITableMetricsData[], contain
         return COLOR_MAP.inactiveOutline;
       }
       return COLOR_MAP.activeOutline;
+    })
+    .on('mouseover', (d) => {
+      const { top, left } = getTooltipPosition(d);
+      onHover(top, left, true, d);
+    })
+    .on('mouseout', (d) => {
+      if (isClicked) {
+        return;
+      }
+      onHover(0, 0, false, d);
+    })
+    .on('click', (d) => {
+      addOverlay(d);
     });
 
+  function getTooltipPosition(d) {
+    const top = y(d.latency) - 70 + 'px';
+    let left = x(d.eventsPerMin) + 50;
```

Comment:
Can you take advantage of `margin` and other constants rather than putting raw numbers here?